### PR TITLE
log refactor: tracer

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
change log pkg to pingca/log

### What is changed and how it works?
update dm/tracer and pkg/tracing

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@amyangfei changed

File: dm/tracer/http_handler.go [hunk 1]
```diff
@@ -62,23 +63,23 @@ func writeError(w http.ResponseWriter, statusCode int, err error) {
...
-		log.Error(errors.ErrorStack(err))
+		log.L().Error("write error", zap.Error(err))
 	}
 }
 
 func writeData(w http.ResponseWriter, data interface{}) {
 	js, err := json.MarshalIndent(data, "", " ")
 	if err != nil {
-		log.Errorf("invalid json data: %v, error: %s", data, err)
+		logger.Error("invalid json data: %v, error: %s", data, err)
```
Comment (GregoryIan) [inline dm/tracer/http_handler.go]: L66 and L77 use `log.L()`, unify them

Comment (GregoryIan) [reply dm/tracer/http_handler.go]: and it must not be ok

File: dm/tracer/http_handler.go [hunk 2]
```diff
@@ -62,23 +63,23 @@ func writeError(w http.ResponseWriter, statusCode int, err error) {
...
 		writeInternalServerError(w, err)
 		return
 	}
-	log.Debug(string(js))
+	log.L().Debug("write data", zap.ByteString("writeData json bytes", js))
```
Comment (GregoryIan) [inline dm/tracer/http_handler.go]:
```suggestion
	log.L().Debug("write data", zap.ByteString("request data", js))
```

File: dm/tracer/http_handler.go [hunk 3]
```diff
@@ -62,23 +63,23 @@ func writeError(w http.ResponseWriter, statusCode int, err error) {
...
 	w.Header().Set(headerContentType, contentTypeJSON)
 	w.WriteHeader(http.StatusOK)
 	_, err = w.Write(js)
 	if err != nil {
-		log.Error(errors.ErrorStack(err))
+		log.L().Error("write data", zap.Error(err))
```
Comment (GregoryIan) [inline dm/tracer/http_handler.go]:
```suggestion
		log.L().Error("fail to write data", zap.Error(err))
```

File: dm/tracer/http_status.go [hunk 4]
```diff
@@ -41,7 +41,7 @@ func (s *Server) startHTTPServer(lis net.Listener) {
 	}
 	err := httpServer.Serve(lis)
 	if err != nil && !common.IsErrNetClosing(err) && err != http.ErrServerClosed {
-		log.Errorf("[server] tracer server return with error %s", err.Error())
+		logger.Error("tracer server return with error", log.ShortError(err))
```
Comment (GregoryIan) [inline dm/tracer/http_status.go]:
```suggestion
		logger.Error("tracer http server return with error", log.ShortError(err))
```

File: dm/tracer/server.go [hunk 5]
```diff
@@ -84,12 +92,12 @@ func (s *Server) Start() error {
...
 		err2 := s.svr.Serve(grpcL)
 		if err2 != nil && !common.IsErrNetClosing(err2) && err2 != cmux.ErrListenerClosed {
-			log.Errorf("[server] gRPC server return with error %s", err2.Error())
+			logger.Error("gRPC server return with error", log.ShortError(err2))
 		}
 	}()
 	go s.startHTTPServer(httpL)
 
-	log.Infof("[server] listening on %v for gRPC API and status request", s.cfg.TracerAddr)
+	logger.Info("listening on tracer address for gRPC API and status request", zap.String("tracer address", s.cfg.TracerAddr))
```
Comment (GregoryIan) [inline dm/tracer/server.go]:
```suggestion
	logger.Info("listening on tracer address for gRPC API and status request", zap.String("address", s.cfg.TracerAddr))
```

File: pkg/tracing/tracer_test.go [hunk 6]
```diff
@@ -27,13 +27,17 @@ import (
...
+	"go.uber.org/zap"
 	"google.golang.org/grpc"
 
 	"github.com/pingcap/dm/dm/common"
 	"github.com/pingcap/dm/dm/pb"
 )
 
 var (
+	// testLogger writes log start with `[component=test-server]`
+	testLogger log.Logger
```
Comment (GregoryIan) [inline pkg/tracing/tracer_test.go]: just use `log.L()`

**Comment by GregoryIan** [comment]
LGTM

**Comment by WangXiangUSTC** [comment]
@csuzhangxc PTAL

File: pkg/tracing/tracer.go [hunk 7]
```diff
@@ -28,6 +28,9 @@ import (
 )
 
 var (
+	// logger writes log start with `[component=tracer]`
+	logger log.Logger
```
Comment (csuzhangxc) [inline pkg/tracing/tracer.go]: where do we set this `component`?

Comment (WangXiangUSTC) [reply pkg/tracing/tracer.go]: no use now, removed

**Pull request merged**